src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs

```diff
@@ -21,28 +24,40 @@ protected DialogWindowHostedToolBase(IUIVisualizerService uiVisualizerService)
     }
 
     public virtual bool IsModal => true;
-    protected override bool StaysOpen => !IsModal;
+    protected override bool StaysOpen => _windowViewModel is not null;
 
-    protected override async Task OnOpenAsync(object? parameter = null)
+    protected override Task OnOpenAsync(object? parameter = null)
     {
-        _parameter = parameter;
+        try
+        {
+            _windowViewModel = InitializeViewModel();
+        }
+        catch (Exception ex)
+        {
+            Log.Error(ex);
+
+            return Task.CompletedTask;
+        }
 
-        _windowViewModel = InitializeViewModel();
-        _windowViewModel.ClosedAsync += OnClosedAsync;
+        _parameter = parameter;
         ApplyParameter(_parameter);
 
         if (IsModal)
         {
-            await _uiVisualizerService.ShowDialogAsync(_windowViewModel, OnWindowCompleted);
+            _uiVisualizerService.ShowDialogAsync(_windowViewModel, OnWindowCompleted);
```

Comment:
I think a modal task should always be awaited?

Reply:
I need this operations not awaited
from here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs4014?f1url=%3FappId%3Droslyn%26k%3Dk(CS4014)
![image](https://github.com/WildGums/Orc.Controls/assets/5163834/0afaa856-2c58-4fd4-8e54-b6d2b7feac4d)

Reply:
why would you not want to await a modal dialog?

Reply:
because i want to finish this function, and keep tool open until window closed...

Reply:
i such case all tools works the same... tools with/without window logic